### PR TITLE
Random port if no port present

### PR DESCRIPTION
**Pull request by cameleeck**

### CODE REVIEW

File: src/main/groovy/com/ofg/stub/util/PortResolver.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Returns port number from service url
+     *
+     * @param url host and port url in format hostname:port
+     *
+     * @return -1 if no port present or port
+     *
+     * @throws IllegalArgumentException when url don't have colon
+     */
+    public static int getPortFromUrlOrRandom(String url) {
```
Comment (nurkiewicz) [inline src/main/groovy/com/ofg/stub/util/PortResolver.java]: What about returning `Optional<Integer>` and then in client code:

    getPortFromUrlOrRandom(url).or(-1)

WDYT? Seems more flexible in that case

File: src/test/groovy/com/ofg/stub/util/PortResolverSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.ofg.stub.util
+
+import spock.lang.Specification
+import com.google.common.base.Optional
+
+class PortResolverSpec extends Specification {
+
+    def "returns IllegarArgumentException when bad format"() {
```
Comment (nurkiewicz) [inline src/test/groovy/com/ofg/stub/util/PortResolverSpec.groovy]: Not sure about this. Should host name without port throw an exception or should we assume *random* port in this case?

File: src/test/groovy/com/ofg/stub/util/PortResolverSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    def "returns correct port"() {
+        when:
+        Optional<Integer> random = PortResolver.getPortFromUrlOrRandom("localhost:2000")
+        then:
+        random.get() == 2000
+    }
+
+    def "return -1 if no port present"() {
```
Comment (nurkiewicz) [inline src/test/groovy/com/ofg/stub/util/PortResolverSpec.groovy]: Actually I believe `localhost:` should be treated as an error while `localhost` alone should mean *random* port. However I am not convinced myself. Any comments?

File: src/main/groovy/com/ofg/stub/util/PortResolver.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+
+    /**
+     * Returns port number from service url
+     *
+     * @param url host and port url in format hostname:port
+     * @return empty if no port present or port
+     * @throws IllegalArgumentException when url don't have colon
+     */
+    public static Optional<Integer> getPortFromUrlOrRandom(String url) {
```
Comment (nurkiewicz) [inline src/main/groovy/com/ofg/stub/util/PortResolver.java]: Rename: `getPortFromUrlOrRandom()` -> `getPortFromUrl()` or `tryGetPortFromUrl()`

File: src/main/groovy/com/ofg/stub/util/PortResolver.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        checkNotNull(url);
+        int colonIndex = url.indexOf(":");
+        checkArgument(colonIndex != -1);
+        String portNumberString = url.substring(colonIndex + 1);
```
Comment (nurkiewicz) [inline src/main/groovy/com/ofg/stub/util/PortResolver.java]: Would:

    org.apache.commons.lang.StringUtils.substringAfterLast(url, ":")

be simpler?

**Comment by nurkiewicz** [comment]
Apart from my comments, looks great. This will make running standalone ZK or multimple microservices on CI server much simpler. **LGTM**.

File: src/test/groovy/com/ofg/stub/util/PortResolverSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.ofg.stub.util
+
+import spock.lang.Specification
+import com.google.common.base.Optional
+
+class PortResolverSpec extends Specification {
+
+    def "returns NullPointerException when null"() {
+        when:
+        PortResolver.tryGetPortFromUrl(null)
```
Comment (szpak) [inline src/test/groovy/com/ofg/stub/util/PortResolverSpec.groovy]: Btw, what do you think about more compact Spock specification formatting?
http://solidsoft.wordpress.com/2014/05/20/more-compact-automatic-spock-specification-formatting-in-idea-13/

**Comment by szpak** [comment]
+1

**Comment by nurkiewicz** [comment]
LGTM, please squash to single commit and merge to master. Thank you!

**Pull request merged**